src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -263,6 +263,17 @@ internal static Conversion GetTrivialConversion(ConversionKind kind)
         internal static ImmutableArray<Conversion> ExplicitEnumerationUnderlying => ConversionSingletons.ExplicitEnumerationUnderlying;
         internal static ImmutableArray<Conversion> PointerToIntegerUnderlying => ConversionSingletons.PointerToIntegerUnderlying;
 
+        // Common combinations to avoid allocations:
+        internal static Conversion ExplicitNullableWithExplicitEnumerationUnderlying = new Conversion(ConversionKind.ExplicitNullable, ExplicitEnumerationUnderlying);
+        internal static Conversion ExplicitNullableWithPointerToIntegerUnderlying = new Conversion(ConversionKind.ExplicitNullable, PointerToIntegerUnderlying);
```

Comment:
I'm somewhat skeptical of this one; how common is it outside a few specific codebases?

Reply:
my view is that it always represents the exact same thing, so we should make it a singleton to match the pattern of everything else.  Having it behave differently is just unnecessary inconsistency *and* allocs when it does appear :)   

The above feels strictly better, especially as it means that the code operates uniformly, and any further updates will continue this pattern to only create new COnversion instances when it truly represents dynamic scenarios :)

Reply:
No.  Conversion is a struct.  So 225-255 always creates the same struct (there's some cpu, but no allocs).  The case of allocs occurs when passing in the child array of conversinos, which then allocates the UncommonData pointer.  

An alternative here would be to cache the UncommonData instances, and then just dynamically construct these structs.  But it would have the same effect, and would require a lot more churn here.  So this approach seems nice and trivial and solves the problem well imo.